pkg/plugin/plugin.go

```diff
@@ -140,8 +142,66 @@ func (pw prefixWriter) Write(buf []byte) (n int, err error) {
 	return len(buf), nil
 }
 
+func getPluginName(archiveName string) string {
+	// if the archive name contains ".app", then it's a macOS app
+	if strings.Contains(archiveName, ".app") {
+		return strings.Split(archiveName, ".app")[0]
+	}
+
+	return strings.Split(archiveName, "_")[0]
+}
+
+func prepareBinary(pluginFilename, pluginPath string) error {
+	pluginName := filepath.Base(pluginPath)
+	binPath := utils.PluginPath(pluginPath, pluginFilename)
+
+	if _, err := os.Stat(binPath); os.IsNotExist(err) {
+		// If the plugin binary does not exist, it means that the plugin is a MacOS .app directory.
+		// No need to rename the binary.
+
+		// We extract the .app.zip file to the plugin directory
+		appPath := filepath.Join(pluginPath, pluginName+".app")
+
+		err = os.Mkdir(appPath, os.ModePerm)
+		if err != nil {
+			return fmt.Errorf("creating the plugin .app directory at %s: %w", appPath, err)
+		}
+
+		appZip := filepath.Join(pluginPath, pluginName+".app.zip")
+
+		err = unzip.Extract(appZip, appPath)
+		if err != nil {
+			return fmt.Errorf("extracting the plugin at %s: %w", appZip, err)
+		}
+
+		return nil
+	}
+
+	newBinPath := utils.PluginPath(pluginPath, pluginName)
+
+	err := os.Rename(filepath.Join(pluginPath, pluginFilename), newBinPath)
+	if err != nil {
+		return fmt.Errorf("renaming plugin %s: %w", pluginName, err)
+	}
+
+	return nil
+}
+
+// binPath returns the path to the plugin binary.
+func (p *Plugin) binPath() string {
+	pluginName := filepath.Base(p.Path)
+
+	binPath := utils.PluginPath(p.Path, pluginName)
+	if _, err := os.Stat(binPath); os.IsNotExist(err) {
+		// Assuming that the plugin is a MacOS .app directory
+		binPath = filepath.Join(p.Path, pluginName+".app", "Contents", "MacOS", pluginName)
```

Comment:
same

Reply:
I would like to keep the code clean and simple. Here it's only used when we run the plugin. So the check of the OS has been done before. I would prefer keep it like that.